### PR TITLE
OTWO-5013: Create new enlistment with same repo but different branch

### PR DESCRIPTION
**Pull request by santhanakarthikeyan**

### CODE REVIEW

File: app/controllers/concerns/enlistment_filters.rb [hunk 1]
```diff
@@ -64,7 +67,36 @@ def validate_project
...
+  def valid_code_location?
+    @code_location = build_code_location
+    @repository = @code_location.repository
+    return render :new, status: :unprocessable_entity unless @code_location.valid?
+  end
+
+  def code_location_exist?
+    code_location = CodeLocation.find_existing(@repository.url, @code_location.module_branch_name)
+    return unless code_location
+    update_repo_username_and_password(code_location)
```
Comment (PDegenPortnoy) [inline app/controllers/concerns/enlistment_filters.rb]: If we find an existing Code Location, why is it important to update the username and password? Were we doing this somewhere else or is this needed functionality that was not previously present?

Comment (notalex) [reply app/controllers/concerns/enlistment_filters.rb]: We were doing it previously in the controller (here|https://github.com/blackducksoftware/ohloh-ui/blob/master/app/controllers/enlistments_controller.rb#L85). This handles the scenario where a code location is present but its repository's username and password have changed.

Comment (PDegenPortnoy) [reply app/controllers/concerns/enlistment_filters.rb]: That's the use case I had assumed.  Very good.

**Comment by PDegenPortnoy** [comment]
We have conflicts that prevent the branch from being merged and the failure in `test/models/reverification_tracker_test.rb:324`

**Comment by santhanakarthikeyan** [comment]
Fixed merge conflicts and Rubocop issues. Merging this PR.

**Pull request merged**